### ISSUE TITLE
Lines wrongly reported as uncovered

### ISSUE THREAD

**Issue by albertored** [erlang/otp#8867]
This is a follow up of https://github.com/elixir-lang/elixir/issues/13863.

I had a problem on elixir where some lines of my code was reported as uncovered also if tests correctly cover them. I was able to reproduce the problem also with plain Erlang code so I'm opening the bug here.

This the module under test

```erlang
-module(cover_bug).

-export([myfun/2]).

myfun(Arg1, <<"bar", _>>) when Arg1 == arg1 orelse Arg1 == arg2 ->
    nil;
myfun(arg3, Arg2) ->
    case lists:sum([10, 2]) of
        12 ->
            Res = Arg2,
            Res
    end.
```

and this is the test

```erlang
?assertEqual(cover_bug:myfun(arg3, <<"foo">>), <<"foo">>)
```

After running that test the second clause of the function is wrongly reported as uncovered. As better explained in the linked Elixir issue almost every change to the code makes the cover report correct again.

**Affected versions**
This happens with OTP 27.1, I've tried also with OTP 26.2.5 and in that case the behaviour is correct.

**Comment by alexslade** [comment]
I'm also hitting this in Elixir running on Erlang 27. (Works as expected on Erlang 26). 

```
defmodule Neg do
  def hello("Prefix." <> name) do
    dbg("Find called with prefix")
    name
  end

  def hello(name) do
    dbg("Find called without prefix")
    "Hello" <> name
  end
end

defmodule NegTest do
  use ExUnit.Case

  test "code covers all lines" do
    assert Neg.hello("Prefix.Alex") == "Alex"
    assert Neg.hello("Alex") == "Hello Alex"
  end
end
```

![Screenshot 2024-10-08 at 20 16 39](https://github.com/user-attachments/assets/cd68148d-4963-4ef2-9121-e94317a9bc60)

**Comment by bjorng** [comment]
Thanks for the bug report.

I've fixed the problem in the linked pull request. I expect it to be included in Erlang/OTP 27.2.
